tests/testlib/s2n_key_schedule_testlib.c

```diff
@@ -0,0 +1,49 @@
+/*
+ * Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License").
+ * You may not use this file except in compliance with the License.
+ * A copy of the License is located at
+ *
+ *  http://aws.amazon.com/apache2.0
+ *
+ * or in the "license" file accompanying this file. This file is distributed
+ * on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+ * express or implied. See the License for the specific language governing
+ * permissions and limitations under the License.
+ */
+
+#include "testlib/s2n_testlib.h"
+
+S2N_RESULT s2n_connection_set_early_secret(struct s2n_connection *conn,
+        const struct s2n_blob *early_secret)
+{
+    RESULT_ENSURE_REF(conn);
+    RESULT_ENSURE_REF(early_secret);
+    RESULT_CHECKED_MEMCPY(conn->secrets.tls13.early_secret,
+            early_secret->data, early_secret->size);
+    conn->secrets.tls13.secrets_count = 1;
```

Comment:
"secrets_count" is essentially how I'm tracking the state of our secret derivation logic. It tracks how far we've gotten.

Alternatives I considered:
* A state variable. It would have the same values as secrets_count, but named things like S2N_SECRETS_INIT, S2N_EARLY_SECRET_READY, S2N_HANDSHAKE_SECRET_READY, S2N_MASTER_SECRET_READY
* A bitfield with flags for early secret, handshake secret, and master secret. Not quite the right model, since most combinations of bitflags wouldn't make much sense.
* Storing blobs alongside the bytes, so that we can just check whether the size is zero or not. I like that this would make the length explicit, but it requires more memory / initialization.